pkg/cmd/pulumi/convert/convert.go

```diff
@@ -457,13 +482,153 @@ func runConvert(
 	return nil
 }
 
-func newPluginContext(cwd string) (*plugin.Context, error) {
-	sink := diag.DefaultSink(os.Stderr, os.Stderr, diag.FormatOptions{
-		Color: cmdutil.GetGlobalColorization(),
-	})
-	pluginCtx, err := plugin.NewContext(sink, sink, nil, nil, cwd, nil, true, nil)
+// getPackagesToGenerateSdks parses the pcl files back in to read the package
+// blocks and for sdk generation.
+func getPackagesToGenerateSdks(
+	sourceDirectory string,
+) (map[string]*schema.PackageDescriptor, hcl.Diagnostics, error) {
+	// TODO use the package block to get the packages and parameterize them and generate SDKs.
+	files, err := os.ReadDir(sourceDirectory)
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not read sourceDirectory: %w", err)
+	}
+
+	parser := hclsyntax.NewParser()
+	_, err = pcl.ParseFiles(parser, sourceDirectory, files)
 	if err != nil {
-		return nil, err
+		return nil, nil, fmt.Errorf("could not parse pcl file: %w", err)
+	}
+
+	allPackageDescriptors := make(map[string]*schema.PackageDescriptor)
+
+	var diagnostics hcl.Diagnostics
+	for _, file := range parser.Files {
+		packageDescriptors, diags := pcl.ReadPackageDescriptors(file)
+		diagnostics = append(diagnostics, diags...)
+		for packageName, descriptor := range packageDescriptors {
+			if _, ok := allPackageDescriptors[packageName]; ok {
+				diagnostics = append(diagnostics, errorf(file.Body.Range(), "package %q was already defined", packageName))
+				continue
+			}
+			allPackageDescriptors[packageName] = descriptor
+		}
+	}
```

Comment:
Wonder if this is worth pulling out into `pcl.ReadAllPackageDescriptors` or similar. Or maybe rename the original to `ReadFilePackageDescriptors` and introduce `ReadFilesPackageDescriptors`? Would de-dupe this with `pcl` and make this smaller again.